pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -130,7 +127,20 @@ func readInstanceID(cfg *VSphereConfig) (string, error) {
 
 	s := object.NewSearchIndex(c.Client)
 
-	svm, err := s.FindByUuid(ctx, dc, strings.ToLower(strings.TrimSpace(out.String())), true, nil)
+	var svm object.Reference
+	for _, v := range addrs {
+		// Strip the cidr
+		ip := strings.Split(v.String(), "/")[0]
```

Comment:
Now that I know net.ParseCIDR() exists, I agree that its wrong :)  I'll push an update for this today.  Thanks @mikedanese